pennylane/gradients/parameter_shift.py

```diff
@@ -1127,8 +1127,8 @@ def circuit(params):
 
     method = "analytic" if fallback_fn is None else "best"
 
-    trainable_params = choose_trainable_params(tape, argnum)
-    diff_methods = find_and_validate_gradient_methods(tape, method, trainable_params)
+    trainable_params_indices = choose_trainable_params(tape, argnum)
+    diff_methods = find_and_validate_gradient_methods(tape, method, trainable_params_indices)

```

Comment:
I would actually revert this change.
`trainable_params` is not the best name, but it is used a lot, for example in the function `choose_trainable_params` and in `tape.trainable_params` :sweat_smile:

Reply:
So for me `trainable_params` indicates the indices into `tape.get_parameters(trainable_only=False)`.  But here we are actually getting indices into the `trainable_param`. 

So if we have:
```
tape = QuantumScript([qml.RX(0.5, 0), qml.RY(1.2, 0), qml.RZ(2.0)], [qml.expval(qml.Z(0)), trainable_params=[1,2])
```

then a `trainable_params_indices =[0]` would correspond to the RY. The zero parameter in trainable params.

I do kinda hate this and find it very confusing, but I don't think we should call it `trainable_params` when it is indices into the trainable params.

Reply:
renamed to `choose_trainable_param_indices`.